### PR TITLE
Throwing init

### PR DESCRIPTION
**Pull request by tonystone**

### CODE REVIEW

File: Pod/Cache/GenericCoreDataStack.swift [hunk 1]
```diff
@@ -108,52 +108,45 @@ public class GenericCoreDataStack<CoordinatorType: NSPersistentStoreCoordinator,
...
                 }
             }
-            NSNotificationCenter.defaultCenter().addObserver(self, selector: "handleContextDidSaveNotification:", name: NSManagedObjectContextDidSaveNotification, object: nil)
-            
-        } catch  let error as NSError {
-            
-            logError(tag) { "Failed to initialize: \(error.localizedDescription)" }
-            return nil
         }
+        NSNotificationCenter.defaultCenter().addObserver(self, selector: #selector(GenericCoreDataStack.handleContextDidSaveNotification(_:)), name: NSManagedObjectContextDidSaveNotification, object: nil)
```
Comment (ndrlndr) [inline Pod/Cache/GenericCoreDataStack.swift:149]: Is listening for notifications still the recommended way to be informed about context changes?

File: Pod/Cache/GenericCoreDataStack.swift [hunk 2]
```diff
@@ -258,7 +258,7 @@ public class GenericCoreDataStack<CoordinatorType: NSPersistentStoreCoordinator,
             //
             if context.parentContext == mainContext {
                 
-                mainContext.performBlockAndWait({ () -> Void in
+                mainContext.performBlock( { () -> Void in
                     do {
                         try self.mainContext.save()
```
Comment (ndrlndr) [inline Pod/Cache/GenericCoreDataStack.swift:263]: Do you need to check whether the mainContext.hasChanges, before saving? I have seen that done elsewhere.

File: Pod/Cache/GenericCoreDataStack.swift (hunk 1 above)
Comment (tonystone) [inline Pod/Cache/GenericCoreDataStack.swift:149]: In this case the only thing we need this for is to save the main context when an edit context is saved so we're not actually listening for changes, just the save and there is no other way to do that.

File: Pod/Cache/GenericCoreDataStack.swift (hunk 2 above)
Comment (tonystone) [inline Pod/Cache/GenericCoreDataStack.swift:263]: Not strictly needed.  The likelihood that there are no changes is low accept in the case where the developer has an edit context and saves for no reason.  Also, I'm limiting non essential changes.  This PR is to change the init to a throwing init in GenericCoreDataStack.

**Pull request merged**